File: ployglots/mta-ployglot-pkg-fetch-client/packages/ts/examples/api/02-client-verb-methods.ts
// @ts-nocheck
/**
 * Example: verb methods on a plain Client.
 *
 * Demonstrates: GET with query params, POST with JSON body, response.json<T>()
 * with a typed shape, and the `Client` alias re-exported as a synonym for
 * `AsyncClient`.
 */
import { Client } from '@polyglot/fetch-http-client';

interface Todo {
  id: number;
  title: string;
  completed: boolean;
}

interface CreatedPost {
  id: number;
  title: string;
  body: string;
  userId: number;
}

async function main(): Promise<void> {
  const client = new Client({ baseUrl: 'https://jsonplaceholder.typicode.com' });
  try {
    const todoResp = await client.get('/todos/1', { params: { _limit: 1 } });
    const todo = await todoResp.json<Todo>();
    console.log('GET /todos/1 →', todo);

    const created = await client.post('/posts', {
      json: { title: 'hello', body: 'from fetch-http-client', userId: 1 },
    });
    const createdData = await created.json<CreatedPost>();
    console.log('POST /posts →', created.statusCode, createdData);
  } finally {
    await client.close();
  }
}

main().catch((err: unknown) => {
  console.error(err);
  process.exitCode = 1;
});

File: ployglots/mta-ployglot-pkg-fetch-client/packages/py/examples/api/02_client_verb_methods.py
"""API example: verb methods with query params, JSON body, typed json()."""

from __future__ import annotations

import asyncio
import sys

from fetch_http_client import AsyncClient


async def main() -> None:
    async with AsyncClient(base_url="https://httpbin.org") as client:
        get_resp = await client.get("/get", params={"q": "fetch-http-client"})
        get_resp.raise_for_status()
        sys.stdout.write(f"GET args={get_resp.json()['args']}\n")

        post_resp = await client.post("/post", json={"name": "polyglot-fetch-http-client"})
        post_resp.raise_for_status()
        sys.stdout.write(f"POST body={post_resp.json()['json']}\n")


if __name__ == "__main__":
    asyncio.run(main())
